tree/tree.h

```diff
@@ -2,6 +2,7 @@
 #define  _MACHINELEARNING_TREE_TREE_H__
 
 #include "splitter.h"
+//#include "criterion.h"
 
 #include <vector>
 
```

Comment:
On a side note, since this is a concrete class, could you provide some examples on how to use it? Should make everything easier on the long run.